thunder/tests/test_interpreter.py

```diff
@@ -1573,6 +1573,56 @@ def fn():
     jit(fn)()
 
 
+def test_enumerate_lookaside(jit):
+    jitting = False
+
+    class mycls:
+        def __init__(self, val):
+            self.list = [val] * 3
+
+        def __iter__(self):
+            assert is_jitting_with_raise() == jitting
+            return self.list.__iter__()
+
+        def __next__(self):
+            assert is_jitting_with_raise() == jitting
+            return self.list.__next__()
```

Comment:
Do we actually hit this `__next__` here?

Reply:
no, I should remove this

Reply:
What about testing with a custom class that does go though next?